### ISSUE TITLE
fix: guard test imports of removed packages with importorskip (C-137)

### ISSUE THREAD

**Issue by Polichinel** [views-platform/views-pipeline-core#106]
## Problem

5 test files unconditionally import packages that were removed from `pyproject.toml` in PR 9. Tests pass locally because `views-reporting` is editable-installed, bringing these as transitive dependencies. In any clean CI environment built from `pyproject.toml` alone, all 5 files fail at **pytest collection time** (not execution) with `ModuleNotFoundError`.

**Affected files and their removed imports:**

| File | Removed packages imported |
|------|--------------------------|
| `tests/test_modules/test_reconciliation.py` | `torch` |
| `tests/test_modules/test_statistics.py` | `torch` |
| `tests/test_modules/test_mapping_characterization.py` | `matplotlib`, `geopandas`, `plotly`, `shapely` |
| `tests/test_modules/test_visualizations.py` | `matplotlib` |
| `tests/test_modules/test_report.py` | `matplotlib` |

**Existing precedent:** `tests/test_modules/test_geopandas_characterization.py` already uses `pytest.importorskip("geopandas")` correctly — tests are skipped gracefully when the package is absent.

### Step 1: Confirm the problem

The falsification tests already exist:
```bash
conda run -n views_pipeline pytest tests/test_falsification_extraction_splash_zone.py::test_h1_no_removed_package_imports -v
```
All 5 should fail.

### Step 2: Replace bare imports with importorskip guards

For each file, replace unconditional top-level imports with `pytest.importorskip`:

**Pattern:**
```python
# Before (fails at collection if package missing):
import torch

# After (skips tests gracefully if package missing):
torch = pytest.importorskip("torch")
```

**For multi-import cases (test_mapping_characterization.py):**
```python
# Before:
import matplotlib
import geopandas
import plotly
from shapely.geometry import box

# After:
matplotlib = pytest.importorskip("matplotlib")
geopandas = pytest.importorskip("geopandas")
plotly = pytest.importorskip("plotly")
shapely = pytest.importorskip("shapely")
from shapely.geometry import box  # safe after importorskip guard
```

**For `import matplotlib.pyplot as plt` (test_visualizations.py, test_report.py):**
```python
# Before:
import matplotlib
import matplotlib.pyplot as plt

# After:
matplotlib = pytest.importorskip("matplotlib")
import matplotlib.pyplot as plt  # safe after importorskip guard
```

### Step 3: Verify falsification tests pass

```bash
conda run -n views_pipeline pytest tests/test_falsification_extraction_splash_zone.py::test_h1_no_removed_package_imports -v
```

All 5 should now pass (the AST parser should see `pytest.importorskip` calls instead of bare `import` statements at the top level).

**Note:** The falsification test (`test_h1_no_removed_package_imports`) uses AST parsing to detect top-level unconditional imports. `pytest.importorskip` is a function call assignment (`torch = pytest.importorskip("torch")`), not an `import` statement — so it won't trigger the check. If needed, update the falsification test to also accept `importorskip` as a valid guard pattern.

### Step 4: Verify the actual tests still pass

```bash
conda run -n views_pipeline pytest tests/test_modules/test_reconciliation.py tests/test_modules/test_statistics.py tests/test_modules/test_mapping_characterization.py tests/test_modules/test_visualizations.py tests/test_modules/test_report.py -v --tb=short
```

All tests that previously passed should still pass (views-reporting is installed locally, so `importorskip` will find the packages and proceed normally).

### Step 5: Full gate

```bash
conda run -n views_pipeline ruff check .
conda run -n views_pipeline pytest -q --deselect tests/test_modules/test_reconciliation.py::TestReconcileCountryWorker
```

(Deselect `TestReconcileCountryWorker` only if issue #105 hasn't been merged yet.)

## Risk register

- **C-137** (Tier 2): 5 test files import removed packages
- Related: C-134 (import path fragility)

## Labels

Bug, extraction, merge-blocker
